spec/integration/use_cases/accounts/apps/evolution_api/webhooks/evolution_apis_controller_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+require 'webmock/rspec'
+require 'sidekiq/testing'
+
+RSpec.describe Apps::EvolutionApisController, type: :request do
+  let(:account) { create(:account) }
+  let(:evolution_api) { create(:apps_evolution_api, account: account) }
+  let(:evolution_api_connected) { create(:apps_evolution_api, :connected, account: account) }
+  let(:qrcode_updated_webhook_event) { load_webhook_event('qrcode_updated_event.json') }
+  let(:created_connection_event) { load_webhook_event('created_connection_event.json') }
+  let(:deleted_connection_event) { load_webhook_event('deleted_connection_event.json') }
+  let(:evolution_api_qrcode_info) { evolution_api.reload.qrcode_info }
+
+  def load_webhook_event(filename)
+    File.read(Rails.root.join('spec/integration/use_cases/accounts/apps/evolution_api/webhooks/events', filename))
+  end
+
+  def post_webhook(event_data)
+    Sidekiq::Testing.inline! { post '/apps/evolution_apis/webhooks', params: JSON.parse(event_data) }
+  end
+
+  def expect_success
+    expect(response).to have_http_status(200)
+    expect(JSON.parse(response.body)).to eq({ 'ok' => true })
+  end
+
+  describe 'POST /apps/evolution_apis/webhooks' do
+    describe 'when is qrcode_update event' do
+      context 'success' do
+        it 'should update qrcode_info' do
+          allow(Time).to receive(:current).and_return(Time.new(2024, 1, 27, 0, 0, 0))
+          evolution_api
+          post_webhook(qrcode_updated_webhook_event)
+          expect_success
+          expect(evolution_api.reload.qrcode_info).to eq({ 'qrcode' => 'qrcode',
+                                                                     'expiration_date' => (Time.new(2024, 1, 27, 0, 0,
+                                                                                                    0) + 50.seconds).to_s })
+        end
+      end
+    end
+    describe 'when is created_connection event' do
+      context 'success' do
+        it 'should update evolution_api status and phone' do
+          evolution_api.update(qrcode_info: { qrcode: 'qrcode', expiration_date: Time.current })
+          post_webhook(created_connection_event)
+          expect_success
+          expect(evolution_api.reload.connection_status).to eq('active')
+          expect(evolution_api.phone).to be_truthy
+          expect(evolution_api.qrcode_info).not_to include('qrcode', 'expired_date')
+        end
+      end
+    end
+    describe 'when is deleted_connection event' do
```

Comment:
Aqui é um caso de conectado e foi desconectado
Acho que é legal adicionar esses dois:
- Conectando e recebe o "close" ( Não leu o qr code)
- Esta desconectado e recebe um evento de qr code ( Esse caso seria em uma situação que os webooks chegam em ordem diferentes)

Reply:
Em ambos os casos, Quando ele está conectado e conectando, ele enviará uma requisição para o evolution_api deletar a instancia, e irá mudar o connection_status para disconnected. Se ele receber o webhook estando disconnected, ele não fará essa requisição, e tbm n mudará o connection_status por já estar disconnected. Na situação de quando ele for receber o close de q não leu o qr code, ele irá cair nessas mesmas regras, pois a estrutura do webhook q recebemos de quando uma pessoa disconecta no celular e quando recebemos o close de q não leu o qr code é igual, só muda o codigo do statusReason.